### DISCUSSION TITLE
Behavior of .fill with multiselect model when initial property value is [] vs null vs ''

### DISCUSSION THREAD

**Discussion by epicwhale** [alpinejs/alpine#4135] [7. Bugs (archived)]
As per the documentation of [.fill](https://alpinejs.dev/directives/model#fill): _"if a bound property is empty, then you can use an input's value attribute to populate the property by adding the .fill modifier"_

While this works for input fields whose value is a null or empty '' string, when I use this with a multiple select field, and if the initialised property value is an empty [] array, it does not seem to replace it with the select's values.

For example:

This sets the `selected_values` state property to [2,3]
```html
<form x-data="{ selected_values: null }">
            <select multiple x-model.fill="selected_values">
                <option value="1">One</option>
                <option value="2" selected>Two</option>
                <option value="3" selected>Three</option>
            </select>
            <div x-text="selected_values"></div>
        </form>
```

This does not work. It sets the `selected_values` state property to [] instead of [1.2]
```html
<form x-data="{ selected_values: [] }">       <!-- note i've made the null to [] -->
            <select multiple x-model.fill="selected_values">
                <option value="1">One</option>
                <option value="2" selected>Two</option>
                <option value="3" selected>Three</option>
            </select>
            <div x-text="selected_values"></div>
        </form>
```

On closer inspection, it looks like the culprit could be this statement which only checks for null and === '', but not an empty array: https://github.com/alpinejs/alpine/blob/2538570d749889db01f98f6975c272ab65a79039/packages/alpinejs/src/directives/x-model.js#L50

I was wondering if this behavior is intended?

**Comment by ekwoka** [comment]
The check itself is as intended.

There is a separate check if you look for checkboxes filling to arrays.

As you know, a Select multiple has fundamentally different behaviors compared to other inputs and may need it's own special check.

I'd probably consider the fact it doesn't work with select multiple to be a bug though.

**Reply by epicwhale** [reply]
You're right, I more meant there perhaps needs to be a more fine-tuned .fill behavior tailored for multiple select, like there's one for input fields today! Unsure, how something makes it from a discussion to bug here, beginner here :) so defer it to the folks here!

**Reply by ekwoka** [reply]
There is no issues. 

You can make a PR to get it checked out and maybe merged

**Comment by bb** [comment]
Should be fixed with https://github.com/alpinejs/alpine/pull/4159. Please give it a try.
